client/search-ui/src/results/StreamingSearchResultsList.tsx

```diff
@@ -203,6 +204,14 @@ export const StreamingSearchResultsList: React.FunctionComponent<
                             as="li"
                         />
                     )
+                default:
+                    // eslint-disable-next-line no-console
+                    console.error(`Unexpected result type, ${result.type}`)
+                    return <Alert className="d-flex m-3" variant="info">
+                        <Text className="m-0">
+                            <strong>Unexpected result type: </strong> <Code>{result.type}</Code>
+                        </Text>
+                    </Alert>
```

Comment:
What is best practice for handling scenarios like this? 

In practice this should never happen in the web app. The value of `result.type` should always be one of the supported values.

Reply:
I think an exhaustive check would be great here.
Then CI would catch it and we won’t ever need to show the Alert.

Like:
```suggestion
                default:
                    const exhaustiveCheck: never = result.type
                    // eslint-disable-next-line no-console
                    console.error(`Unexpected result type, ${exhaustiveCheck as string}`)
                    return <></>
```

Reply:
Thanks for the suggestion! I'm getting this build-ts error: `Type 'string' is not assignable to type 'never'.` TS noob here, is there a way around this to achieve what you're suggesting?

Reply:
@tbliu98 Exhaustive types only work for union types. If TypeScript can't enumerate all possible values (and with `string` that is impossible lol) this doesn't work :(